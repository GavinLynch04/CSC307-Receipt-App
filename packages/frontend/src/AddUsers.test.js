import React from 'react';
import TestRenderer from 'react-test-renderer';
import InputField from './AddUsers.js';

//Make sure CSS import in AddUsers.js is commented out
describe('InputField component', () => {
    it('renders correctly', () => {
        const tree = TestRenderer.create(<InputField onInputChange={() => {}} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles input change and form submission', () => {
        // Mock the onInputChange function
        const mockOnInputChange = jest.fn();

        // Create a test renderer instance
        const renderer = TestRenderer.create(<InputField onInputChange={mockOnInputChange} />);
        const instance = renderer.root;  // Use .root instead of .getInstance()

        // Simulate user input
        instance.findByType('input').props.onChange({ target: { value: 'TestInput' } });

        // Trigger form submission
        instance.findByType('form').props.onSubmit({ preventDefault: jest.fn() });

        // Check if onInputChange was called with the correct value
        expect(mockOnInputChange).toHaveBeenCalledWith('TestInput');
    });
});
