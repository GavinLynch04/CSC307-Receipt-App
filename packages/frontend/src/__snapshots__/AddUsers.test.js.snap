// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component renders correctly 1`] = `
<form
  className="userForm"
  onSubmit={[Function]}
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <input
    className="username"
    onChange={[Function]}
    placeholder="Input a name to update or add new buttons"
    type="text"
    value=""
  />
  <button
    className="submitButton"
    type="submit"
  >
    Submit
  </button>
</form>
`;
